core/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java

```diff
@@ -96,15 +97,16 @@ protected boolean transportCompress() {
         private final NodesRequest request;
         private final String[] nodesIds;
         private final ActionListener<NodesResponse> listener;
-        private final ClusterState clusterState;
+        ImmutableOpenMap<String, DiscoveryNode> nodes;
```

Comment:
why not just resolve this at the start method and not keep this variable as well?